src/cpp/src/text_callback_streamer.hpp

```diff
@@ -9,20 +9,23 @@
 namespace ov {
 namespace genai {
 
-class TextCallbackStreamer: public StreamerBase {
+class TextCallbackStreamer : public StreamerBase {
 public:
-    bool put(int64_t token) override;
-    void end() override;
-
     TextCallbackStreamer(const Tokenizer& tokenizer, std::function<bool(std::string)> callback);
 
-    std::function<bool(std::string)> on_finalized_subword_callback = [](std::string words)->bool { return false; };
+    bool put(int64_t token) override;
```

Comment:
should it override `put(vector)` to have optimal implementation?

Reply:
`TextCallbackStreamer` cannot work with `put(std::vector<int64_t>)` properly for the moment. The trick with inserting chunk of tokens to streamer's cache looks not valid to me. This is due to `n_delay_tokens` feature: https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/text_callback_streamer.cpp#L42.
I propose to fix it in the next PRs.